src/Sarif/Core/PropertyBagHolder.cs

```diff
@@ -99,7 +99,26 @@ public T GetProperty<T>(string propertyName)
                         propertyName));
             }
 
-            return JsonConvert.DeserializeObject<T>(Properties[propertyName].SerializedValue);
+            SerializedPropertyInfo propValue = Properties[propertyName];
+            if (propValue == null)
+            {
+                if (typeof(T).IsValueType)
+                {
+                    throw new InvalidOperationException(
+                        string.Format(
+                            CultureInfo.CurrentCulture,
+                            SdkResources.PropertyOfValueTypeCannotBeNull,
+                            propertyName,
+                            typeof(T).FullName));
+                }
+
+                // This will return null for reference types. Could not set null here b/c T could be a value type as well.
```

Comment:
>well [](start = 117, length = 4)

LOL! This makes me happy, because I was sad when the `default` literal didn't appear in my proposed code :-)